### ISSUE TITLE
Failure with scripts_regression_tests from my mac: could not initialize machine object

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#2215]
I have a `.cime` directory on my mac whose contents are here: https://github.com/billsacks/mac_cime_configuration

Up through 05ba5c55f I am able to run `./scripts_regression_tests.py A_RunUnitTests`. But starting with 7e28a2152 (`Merge pull request #2148 from ESMCI/jgfouca/hide_xml_interface`), I get errors. In that commit, the error is:

```
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 2607, in <module>
    _main_func()
  File "./scripts_regression_tests.py", line 2534, in _main_func
    MACHINE = Machines()
  File "/Users/sacks/cime/scripts/tests/../lib/CIME/XML/machines.py", line 49, in __init__
    machine = self.probe_machine_name()
  File "/Users/sacks/cime/scripts/tests/../lib/CIME/XML/machines.py", line 103, in probe_machine_name
    machine = self._probe_machine_name_one_guess(nametomatch)
  File "/Users/sacks/cime/scripts/tests/../lib/CIME/XML/machines.py", line 128, in _probe_machine_name_one_guess
    nodes = self.get_children("machine")
  File "/Users/sacks/cime/scripts/tests/../lib/CIME/XML/generic_xml.py", line 183, in get_children
    assert validate == children, "Validation failed for {}, {}\nScan found {} elements, get_children found {}".format(name, attributes, len(validate), len(children))
AssertionError: Validation failed for machine, None
Scan found 30 elements, get_children found 29
```

In master, the error is:

```
No handlers could be found for logger "CIME.XML.machines"
ERROR: Could not initialize machine object from /Users/sacks/cime/config/cesm/machines/config_machines.xml or /Users/sacks/.cime/config_machines.xml
```

Does this indicate a problem in cime, or something that I need to update in my own cime configuration?

**Comment by jgfouca** [comment]
@billsacks , I believe I turned off validation in a recent PR. Can you pull and try again?

**Comment by billsacks** [comment]
With latest master, I'm getting the second error:

```
No handlers could be found for logger "CIME.XML.machines"
ERROR: Could not initialize machine object from /Users/sacks/cime/config/cesm/machines/config_machines.xml or /Users/sacks/.cime/config_machines.xml
```

which maybe is completely different and the title of this issue may need to be changed.

**Comment by billsacks** [comment]
Indeed, I see that the original "scan" error disappeared in

```
89015ca13 Merge pull request #2205 from ESMCI/jgfouca/remove_scans
```

and the new error started at that point. That either means:

(1) PR #2205 fixed the first problem and caused the second

or

(2) The second problem was introduced sometime since 05ba5c5 but was masked by the first problem until it was fixed.

**Comment by billsacks** [comment]
I tried rebasing the changes from PR #2205 onto 7e28a21 (#2148 ) and I still get the second error ("Could not initialize machine object"). So either #2205 introduced this error, or this error was introduced in #2148 but was masked by the first error until now.

**Comment by jedwards4b** [comment]
@billsacks  I think that in .cime/config_machines.xml  the
<config_machines>
</config_machines>
puts this machine one level deeper than the other machines, can you try again after removing those two lines...

**Comment by billsacks** [comment]
@jedwards4b thank you - you're right. When I remove those lines, it works. I also removed the similar lines from my config_compilers, although I'm not sure that's needed.

I'll close this issue, but it looks like `userdefined_laptop_template` needs to be fixed; I'll open a separate issue for that.

**Comment by lvankampenhout** [comment]
Just wanted to let you know that I ran into the same error (Could not initialize machine object...) using the template at $CIME/config/xml_schemas/config_machines_template.xml  

and following the steps outlined here: http://esmci.github.io/cime/users_guide/porting-cime.html?highlight=pnetcdf#steps-for-porting

Either the documentation should be updated ("remove <config_machines> tags") or the template.

**Comment by lvankampenhout** [comment]
This resolved it for me:
export CIME_MACHINE=cartesius
./scripts_regression_tests.py

This may have to do with the fact that I didn't define NODENAME_REGEX so the script is in the dark on which machine it is executing.

**Comment by billsacks** [comment]
@lvankampenhout yes, it makes sense to me that you'd either need to set the machine name explicitly or make NODENAME_REGEX able to pick it up.

**Comment by billsacks** [comment]
@lvankampenhout FYI after #2272 is merged to master, it sounds like you'll need to add the outer-level `<config_machines>` tags back in.

**Comment by lvankampenhout** [comment]
thanks for the heads-up :)

**Comment by jedwards4b** [comment]
@billsacks to be clear the change is backward compatible and you don't need the outermost tag, however its better to have it than not.
